crates/routes/src/images.rs

```diff
@@ -253,20 +250,32 @@ async fn delete(
 pub async fn image_proxy(
   Query(params): Query<ImageProxyParams>,
   req: HttpRequest,
-  client: web::Data<ClientWithMiddleware>,
-  context: web::Data<LemmyContext>,
+  client: Data<ClientWithMiddleware>,
+  context: Data<LemmyContext>,
 ) -> LemmyResult<HttpResponse> {
   let url = Url::parse(&params.url)?;
 
   // Check that url corresponds to a federated image so that this can't be abused as a proxy
   // for arbitrary purposes.
   RemoteImage::validate(&mut context.pool(), url.clone().into()).await?;
 
-  let pictrs_config = context.settings().pictrs_config()?;
-
-  let processed_url = params.process_url(&params.url, &pictrs_config.url);
-
-  image(processed_url, req, &client).await
+  let bypass_proxy = context
+    .settings()
+    .pictrs_config()?
+    .proxy_bypass_domains
+    .iter()
+    .any(|s| url.domain().is_some_and(|domain| domain.starts_with(s)));
+  if bypass_proxy {
+    Ok(
+      Redirect::to(url.to_string())
+        .respond_to(&req)
+        .map_into_boxed_body(),
+    )
+  } else {
+    let pictrs_config = context.settings().pictrs_config()?;
```

Comment:
Why is the processed_url removed now?

Reply:
Must have been lost while handling conflicts. Fixed now.